crates/ruff_linter/src/rules/pycodestyle/rules/logical_lines/continuation_lines.rs

```diff
@@ -0,0 +1,803 @@
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Closing bracket not matching its corresponding opening bracket's indentation.")
+    }
+}
+
+/// ## What it does
+/// Checks for closing brackets that do not match the indentation of the opening bracket.
+///
+/// ## Why is this bad?
+/// This makes identifying brackets pair harder.
+///
+/// ## Example
+/// ```python
+/// result = function_that_takes_arguments('a', 'b', 'c',
+///                                        'd', 'e', 'f',
+/// )
+/// ```
+///
+/// Use instead:
+/// ```python
+/// result = function_that_takes_arguments('a', 'b', 'c',
+///                                        'd', 'e', 'f',
+///                                       )
+/// ```
+///
+/// [PEP 8]: https://www.python.org/dev/peps/pep-0008/#indentation
+#[violation]
+pub struct ClosingBracketNotMatchingOpeningBracketVisualIndentation;
+
+impl Violation for ClosingBracketNotMatchingOpeningBracketVisualIndentation {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!(
+            "Closing bracket not matching its corresponding opening bracket's visual indentation."
+        )
+    }
+}
+
+/// ## What it does
+/// Checks for continuation lines with the same indent as the next logical line.
+///
+/// ## Why is this bad?
+/// Continuation lines should not be indented at the same level as the next logical line.
+/// Instead, they should be indented to one more level so as to distinguish them from the next line.
+///
+/// ## Example
+/// ```python
+/// if user is not None and user.is_admin or \
+///     user.name == 'Grant':
+///     blah = 'yeahnah'
+/// ```
+///
+/// Use instead:
+/// ```python
+/// if user is not None and user.is_admin or \
+///         user.name == 'Grant':
+///     blah = 'yeahnah'
+/// ```
+///
+/// [PEP 8]: https://www.python.org/dev/peps/pep-0008/#indentation
+#[violation]
+pub struct ContinuationLineIndentSameAsNextLogicalLine;
+
+impl Violation for ContinuationLineIndentSameAsNextLogicalLine {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Continuation line with same indent as next logical line.")
+    }
+}
+
+/// ## What it does
+/// Checks for continuation line over-indented for hanging indent.
+///
+/// ## Why is this bad?
+/// This makes distinguishing continuation lines harder.
+///
+/// ## Example
+/// ```python
+/// print("Python", (
+///         "Rules"))
+/// ```
+///
+/// Use instead:
+/// ```python
+/// print("Python", (
+///     "Rules"))
+/// ```
+///
+/// [PEP 8]: https://www.python.org/dev/peps/pep-0008/#indentation
+#[violation]
+pub struct ContinuationLineOverIndentedForHangingIndent;
+
+impl Violation for ContinuationLineOverIndentedForHangingIndent {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Continuation line over indented for hanging indent.")
+    }
+}
+
+/// ## What it does
+/// Checks for continuation line over-indented for visual indent.
```

Comment:
The difference between this and `ContinuationLineOverIndentedForHangingIndent` and `ContinuationLineUnderIndentedForVisualIndent` isn't clear to me from reading the documentation. Are these different styles and users should only enable one of those rules?

Reply:
It seems to me like they could both apply in one codebase, although that would not be a very consistent codebase.

Looking at the [fixtures from pycodestyle](https://github.com/PyCQA/pycodestyle/blob/main/testing/data/E12.py),  `E126` (`ContinuationLineOverIndentedForHangingIndent`) seems to apply when there is a newline after the symbol causing the continuation line (`(`, `\`, etc...) whereas `E128` (`ContinuationLineUnderIndentedForVisualIndent`) applies when there is at least one value after the start of the continuation line.

For example this is a `E126` error
```python
print("E126", (
                "1",
    "2",
))
```

Whereas this is a `E128` error:
```python
print("E128", ("1",
    "2",
))
```

See also the unofficial rules explanation for [E126](https://www.flake8rules.com/rules/E126.html) and [E128](https://www.flake8rules.com/rules/E128.html)